src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+type Colors = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape.Triangle | Shape.Circle | Shape.Rectangle;
+  color: Colors;
 
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [this.a, this.b, this.c];
+
+    sides.sort((x: number, y: number) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('Error');
+    }
+
+    if (sides.some((el) => el <= 0)) {
+      throw new Error('Error');
+    }
+  }
 
+  getArea(): number {
```

Comment:
```suggestion
  getArea() {
```
As your class implements `Figure` there's no need to specify it.